### PR TITLE
RFC - Loop Optimization

### PR DESCRIPTION
**Pull request by jmoyerman**

### CODE REVIEW

File: src/Mount.cpp [hunk 1]
```diff
@@ -3039,12 +3033,15 @@ void Mount::loop()
...
                 // Make sure we do one last update when the steppers have stopped.
+                // TODO this should probably be throttled, 250ms won't make a difference here.
                 displayStepperPosition();
             }
         }
     }
 
     _stepperWasRunning = raStillRunning || decStillRunning;
+
+    displayStepperPositionThrottled();
```
Comment (ClutchplateDude) [inline src/Mount.cpp:3044]: Why did you add this?

File: src/b_setup.hpp [hunk 2]
```diff
@@ -47,23 +45,16 @@ WifiControl wifiControl(&mount, &lcdMenu);
 //   Interrupt handling
 /////////////////////////////////
 /* There are three possible configurations for periodically servicing the steper drives:
- * 1) If RUN_STEPPERS_IN_MAIN_LOOP != 0 then the stepper drivers are called from Mount::loop().
- *    Performance depends on how fast Mount::loop() can execute. With serial or UI activity it 
- *    is likely that steps will be missed and tracking may not be smooth. No interrupts or threads
- *    are used, so this is simple to get running.
- * 2) If ESP32 is #defined then a periodic task is assigned to Core 0 to service the steppers.
+ * 1) If ESP32 is #defined then a periodic task is assigned to Core 0 to service the steppers.
```
Comment (ClutchplateDude) [inline src/b_setup.hpp:51]: There's only two now :-)

File: src/Mount.cpp (hunk 1 above)
Comment (jmoyerman) [reply src/Mount.cpp:3044]: It was moved from loop() in c_buttons to here. Trying to consolidate how many calls are coming from loop() into each Service/Class

File: src/c_buttons.hpp [hunk 3]
```diff
@@ -280,6 +280,15 @@ void wifiLoop() {
 }
 #endif
 
+#if !defined(OAT_DEBUG_BUILD) && defined(SUPPORT_SERIAL_CONTROL)
+void serialLoop() {
+    meadeSerialStreamReader.loop();
+}
+#else
+void serialLoop() {
+
```
Comment (ClutchplateDude) [inline src/c_buttons.hpp:289]: No serial in a DEBUG build?

File: src/libs/MeadeStreamReader/MeadeStreamReader.cpp [hunk 4]
```diff
@@ -3,30 +3,34 @@
...
+    this->newData  = false;
+    this->i        = 0;
 }
 
 
-void MeadeStreamReader::loop() {
-    if (stream) // TODO error on not init?
+void MeadeStreamReader::loop()
+{
+    FastPin<7>::high();
```
Comment (ClutchplateDude) [inline src/libs/MeadeStreamReader/MeadeStreamReader.cpp:29]: I assume this is for debugging with a logic analyzer? It should be in #ifdef DEBUG_SOMETHING unless you're planning to rip it all out when you're done.

File: Configuration.hpp [hunk 5]
```diff
@@ -39,7 +39,7 @@
 // Uncomment the definition and set a board (see Constants.hpp for valid values) if you build in Arduino IDE.
 // PlatformIO will set this value automatically and no action is needed.
 #ifndef BOARD
-// #define BOARD BOARD_AVR_MEGA2560
+#define BOARD BOARD_AVR_MEGA2560
```
Comment (ClutchplateDude) [inline Configuration.hpp:42]: I think this causes issues with PIO....

File: src/libs/MeadeStreamReader/MeadeStreamReader.hpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <Arduino.h>
+
+const byte NUM_CHARS = 32;
```
Comment (ClutchplateDude) [inline src/libs/MeadeStreamReader/MeadeStreamReader.hpp:5]: I think :XGM# might return more than 32....

File: src/libs/MeadeStreamReader/MeadeStreamReader.cpp (hunk 4 above)
Comment (jmoyerman) [reply src/libs/MeadeStreamReader/MeadeStreamReader.cpp:29]: Yeah. I'll be pulling this out before merge.

File: src/c_buttons.hpp (hunk 3 above)
Comment (jmoyerman) [reply src/c_buttons.hpp:289]: This was existing functionality that I just moved over. I think the debugger takes over uart0 anyway

File: Configuration.hpp (hunk 5 above)
Comment (jmoyerman) [reply Configuration.hpp:42]: Probably. This was a whoops of me being dumb.

File: src/libs/MeadeStreamReader/MeadeStreamReader.hpp (hunk 6 above)
Comment (jmoyerman) [reply src/libs/MeadeStreamReader/MeadeStreamReader.hpp:5]: This is a receive only buffer

Comment (ClutchplateDude) [reply src/libs/MeadeStreamReader/MeadeStreamReader.hpp:5]: Ahh, doh, of course....

**Comment by jmoyerman** [comment]
Going to split these up a bit.
